lib/RunTests.js

```diff
@@ -164,7 +200,23 @@ function runTests(
       const mainModule = Generate.getMainModule(project.generatedCodeDir);
       const dest = path.join(project.generatedCodeDir, 'elmTestOutput.js');
 
-      Generate.generateElmJson(project);
+      await Generate.generateElmJson(project, (progress) => {
```

Comment:
I’ve never used async interators, but don’t they imply that if you do `break` in the `for await` loop, the job will be cancelled/stopped at that point? Or, like, that the consumer is “in charge”. `getExecutable` doesn’t really work that way (but maybe it could, if there’s need for it, I guess it’s possible to kill a spawned process (such as `curl`, which elm-tooling uses for downloading) midway) – it gives you a `Promise` that (might) kick of a download, and you can’t cancel it. You can’t control it during the way – all you get is a “notification” about progress updates every now and then.

Reply:
Yeah okay, I buy the "notification" and "control" arguments that callbacks are the most appropriate thing to use here.